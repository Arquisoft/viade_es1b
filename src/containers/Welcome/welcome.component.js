import React from 'react';
import { LoggedOut, LoggedIn, useWebId } from '@solid/react';
import { Redirect } from 'react-router-dom';
import { Trans, useTranslation } from 'react-i18next';
import {
  WelcomeCard,
  WelcomeLogo,
  WelcomeProfile,
  WelcomeDetail
} from './welcome.style';

/**
 * Welcome Page UI component, containing the styled components for the Welcome Page
 * Image component will get theimage context and resolve the value to render.
 * @param props
 */
export const WelcomePageContent = props => {
  const { t } = useTranslation();
  const name = useWebId();
  console.log(name);

  const style = {
    position: 'absolute',
    width: '100%',
    height: '92.4%',
    backgroundImage: 'linear-gradient(to right, white, lightblue)',
    backgroundRepeat: 'repeat',
    padding: '50px 0',
  };
  return (
    <section data-testid="welcome-wrapper" style={style}>
      <LoggedIn>
        <WelcomeCard className="card">
          <WelcomeLogo data-testid="welcome-logo">
            <img src={process.env.PUBLIC_URL + "/img/inrupt.svg"} alt="Inrupt" />
          </WelcomeLogo>
          <WelcomeProfile data-testid="welcome-profile">
            <h3>
              {t('welcome.welcome')}, <a href={name}>{name}</a>
            </h3>
          </WelcomeProfile>
        </WelcomeCard>
        <WelcomeCard className="card">
          <WelcomeDetail data-testid="welcome-detail">
            <Trans i18nKey="welcome.title">
              <h3>
                title
            </h3>
            </Trans>

            <p>{t('welcome.summary')}</p>
            <h3>{t('welcome.contactUsTitle')}</h3>
            <Trans i18nKey="welcome.contactUsText">
              <p>
                If you have additional questions about the use of the Application, you can go to our
              <a href="https://github.com/Arquisoft/viade_es1b">GitHub site</a>.
            </p>
            </Trans>
          </WelcomeDetail>
        </WelcomeCard>
      </LoggedIn>
      <LoggedOut>
        <Redirect to='/login'></Redirect>
      </LoggedOut>
    </section>
  );
};
