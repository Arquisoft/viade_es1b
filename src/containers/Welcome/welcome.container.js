import React from "react";
import { WelcomePageContent } from "./welcome.component";
const defaultProfilePhoto = "/img/icon/empty-profile.svg";

/**
 * Container component for the Welcome Page, containing example of how to fetch data from a POD
 */
export class WelcomeComponent extends React.Component<Props> {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      image: defaultProfilePhoto,
      isLoading: false,
      hasImage: false,
    };
  }

  render() {
    const { webId } = this.props;
    return (
      <WelcomePageContent
        {...{ webId }}
      />
    );
  }
}
