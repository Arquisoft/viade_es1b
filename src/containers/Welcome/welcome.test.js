import React from "react";
import { render, queryByTestId, act } from "@testing-library/react";
import Welcome from "../Welcome";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() =>
  act(() => {
    const { container, debug } = render(<Welcome />);
    wrapper = container;
    //debug();
  })
);

describe("Welcome Page Render", () => {
  test("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  test("renders all components", () => {
    expect(queryByTestId(wrapper, "welcome-logo")).not.toBeNull();
    expect(queryByTestId(wrapper, "welcome-wrapper")).not.toBeNull();
    expect(queryByTestId(wrapper, "welcome-profile")).not.toBeNull();
  });

  test("renders with default props", () => {
    const wrappers = shallow(<Welcome />);
    expect(wrappers).toMatchSnapshot();
  });
});
