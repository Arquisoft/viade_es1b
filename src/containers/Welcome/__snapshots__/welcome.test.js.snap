// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page Render renders with default props 1`] = `ShallowWrapper {}`;
